GG/GG/TextControl.h

```diff
@@ -74,9 +74,42 @@ class GG_API TextControl : public Control
     TextControl(X x, Y y, X w, Y h, const std::string& str, const boost::shared_ptr<Font>& font,
                 Clr color = CLR_BLACK, Flags<TextFormat> format = FORMAT_NONE,
                 Flags<WndFlag> flags = NO_WND_FLAGS);
+
+    /** Fast constructor.
+
+     This constructor requires a \p str and \text_elements that are consistent with each other.
+     Font::ExpensiveParseFromTextToTextElements() will not be called on \p str.  Hence this
+     constructor is much faster than the first constructor.*/
+    TextControl(X x, Y y, X w, Y h, const std::string& str,
+                const std::vector<boost::shared_ptr<Font::TextElement> >&text_elements,
+                const boost::shared_ptr<Font>& font,
+                Clr color = CLR_BLACK, Flags<TextFormat> format = FORMAT_NONE,
+                Flags<WndFlag> flags = NO_WND_FLAGS);
+
+    /** Copy constructor.
+
+        Text Control requires a copy-constructor because m_text_elements contains pointers
+        to m_text which need to be bound with Bind() to the m_text in the new TextControl.
+
+        Using the copy constructor is faster than constructing a TextControl with text from another
+        TextControl because it avoids the XML parse overhead.
+
+        Since Control does not have a way to access Flags the copy using default flags.
```

Comment:
using -> will use

Perhaps you should add a copy constructor to Wnd and Control to copy the flags with then?